Framework/PythonInterface/mantid/plots/__init__.py

```diff
@@ -1045,6 +1048,217 @@ def tricontourf(self, *args, **kwargs):
         """
         return self._plot_2d_func('tricontourf', *args, **kwargs)
 
+    def set_initial_dimensions(self):
+        # Set the width and height which are used to calculate the offset percentage for waterfall plots.
+        # This means that the curves in a waterfall plot are always offset by the same amount, even if the
+        # plot limits change.
+        x_lim, y_lim = self.get_xlim(), self.get_ylim()
+        self.width = x_lim[1] - x_lim[0]
+        self.height = y_lim[1] - y_lim[0]
+
+    def is_waterfall_plot(self):
+        return self.waterfall_x_offset != 0 or self.waterfall_y_offset != 0
+
+    def update_waterfall_plot(self, x_offset, y_offset):
+        errorbar_cap_lines = self.remove_and_return_errorbar_cap_lines()
+
+        for i in range(len(self.get_lines())):
+            self.convert_single_line_to_waterfall(i, x_offset, y_offset)
+
+        # Now set the new offsets
+        self.waterfall_x_offset = x_offset
+        self.waterfall_y_offset = y_offset
+
+        if any(isinstance(collection, PolyCollection) for collection in self.collections):
+            self.waterfall_update_fill()
+
+        self.lines += errorbar_cap_lines
+
+        self.set_waterfall_toolbar_options_enabled()
+        self.get_figure().canvas.draw()
+
+    def set_waterfall_toolbar_options_enabled(self):
+        toolbar = self.get_figure().canvas.toolbar
+        toolbar.waterfall_conversion(self.is_waterfall_plot())
+
+    def convert_to_waterfall(self):
+        if self.is_waterfall_plot():
+            return
+
+        # Plots made from a script may not have the width and height attributes needed to create a waterfall plot, so
+        # they are set here.
+        if not hasattr(self, 'width'):
+            self.set_initial_dimensions()
+
+        # 10 and 20 are the default x and y offset.
+        self.update_waterfall_plot(10, 20)
```

Comment:
Can we define "magic" numbers at module scope. This gives them a name and makes the code easier to read so that you wouldn't need the comment above. Something like
```python
WATERFALL_XOFFSET_DEFAULT, WATERFALL_YOFFSET_DEFAULT = 10, 20
```
?